drivers/serial/Kconfig.nrfx

```diff
@@ -33,6 +33,7 @@ config UART_NRFX_UARTE_LEGACY_SHIM
 	depends on UART_NRFX_UARTE
 	depends on !SOC_SERIES_NRF54LX
 	depends on RISCV || !SOC_SERIES_NRF54HX
+	depends on RISCV || !SOC_SERIES_NRF92X
```

Comment:
Is this any different from oneliner?

`depends on !(SOC_SERIES_NRF54HX || SOC_SERIES_NRF92X) || RISCV`

At least I find the current two lines confusing.
The oneliner above I understand as "this is unavailable on nRF54L".
This one I proposed I read as "this is unavailable on the SoCs from the list (54H, 92) for other CPUs than RISC-Vs".

Ok, maybe by changing the order it would be easier to understand:

```
depends on !SOC_SERIES_NRF54HX || RISCV
depends on !SOC_SERIES_NRF92X || RISCV
```
meaning "unavailable on 54H unless in VPR" and "... nRF92 unless in VPR"